leetcode/581.shortest-unsorted-continuous-subarray.py

```diff
@@ -0,0 +1,28 @@
+class Solution(object):
+    def findUnsortedSubarray(self, nums):
+        """
+        :type nums: List[int]
+        :rtype: int
+        """
+        n = len(nums)
+        l, i = 0, 0
+        for i in range(1, n):
+            if nums[i]<nums[i-1]:
+                l = i
+                break
+
+        for i in range(i, n):
+            while l>=0 and nums[i]<nums[l]:
+                l-=1
+
+        r, i = n-1, n-1
+        for i in range(n-2, -1, -1):
+            if nums[i]<nums[i+1]:
+                r = i
+                break
+
+        for i in range(i, -1, -1):
+            while r<n and nums[i]>nums[r]:
+                r+=1
+
+        return r-l+1
```

Comment:
```
Input: [1]
Output: 1
Expected: 0
```
End:
1. usually r>l
2. r==l: one element, [1], return 0
3. r<l: no trend reverse [1,2,3,4], r and l keeps at the initial position